File: samples/legacy-architecture-samples/simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/application/BaseFragment.java
package com.zhuinden.simplestackexamplemvvm.application;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

/**
 * Created by Zhuinden on 2017.07.26..
 */

public abstract class BaseFragment<VM>
        extends Fragment {
    @NonNull
    public final <T extends BaseKey> T getKey() {
        Bundle args = getArguments();
        if(args == null) {
            throw new IllegalStateException("Fragment cannot have null arguments.");
        }
        T key = args.getParcelable("KEY");
        if(key == null) {
            throw new IllegalStateException("Fragment cannot have null key");
        }
        return key;
    }

    public abstract void bindViewModel(VM viewModel);
}

File: samples/multistack-samples/simple-stack-example-multistack-fragment/src/main/java/com/zhuinden/simplestackdemomultistack/core/navigation/BaseFragment.kt
package com.zhuinden.simplestackdemomultistack.core.navigation

import androidx.fragment.app.Fragment

/**
 * Created by Zhuinden on 2018.09.17.
 */

open class BaseFragment : Fragment() {
    fun <T : MultistackFragmentKey> getKey(): T = arguments!!.getParcelable<T>("KEY")!!
}
